### ISSUE TITLE
[icon] enable semantics on host element while removing them from inner SVG

### ISSUE THREAD

**Issue by samrichardsontylertech** [tyler-technologies-oss/forge#395]
**Describe the feature:**
By default the `<forge-icon>` component hides itself from assistive technology by setting `aria-hidden="true"` on the host element. This is fine for purely decorative icons but requires setting `aria-hidden="false"` if the icon should have semantic meaning, which functions differently from just not setting `aria-hidden` and is somewhat of an antipattern. It also adds an attribute by default which should be avoided as much as possible.

`aria-hidden="true"` should be set on the inner SVG element instead, avoiding these problems while keeping the same semantics by default and providing a smoother surface for adding `role="img"` and an `alt` attribute or label to the component if needed.

**Describe any alternatives you've considered:**
Setting `aria-hidden="false"` on the `<forge-icon>` element exposes it to assistive technology but has the drawbacks listed above.
